src/vm/core.zig

```diff
@@ -673,6 +677,40 @@ pub const CairoVM = struct {
         return null;
     }
 
+    // Relocates the VM's trace, turning relocatable registers to numbered ones
+    pub fn relocateTrace(self: *Self, relocation_table: []usize) !void {
+        if (self.trace_relocated) {
+            return TraceError.AlreadyRelocated;
+        }
+        if (relocation_table.len < 2) {
+            return TraceError.NoRelocationFound;
+        }
+        switch (self.trace_context.state) {
+            .enabled => |trace_enabled| {
+                for (trace_enabled.entries.items) |entry| {
+                    try self.trace_context.addRelocatedTrace(.{
+                        .pc = Felt252.fromInteger(try entry.pc.relocateAddress(relocation_table)),
+                        .ap = Felt252.fromInteger(try entry.ap.relocateAddress(relocation_table)),
+                        .fp = Felt252.fromInteger(try entry.fp.relocateAddress(relocation_table)),
```

Comment:
That is strange but this is not exactly what is implemented in Rust in my opinion. I see that in Python we have exactly what you implemented here and what seems logical to me because you relocate all the registers in the same way: https://github.com/starkware-libs/cairo-lang/blob/51c287ac441ea0295775dce743470f87e395d4b1/src/starkware/cairo/lang/vm/trace_entry.py#L62-L68

On the other hand in Rust the implementation seems different with all the segments which are relocated to numbered ones apart from the pc: https://github.com/lambdaclass/cairo-vm/blob/8887e60f7ea5459188cb6624acc908ad29e75f60/vm/src/vm/runners/cairo_runner.rs#L778-L780

I can't understand why this difference because it is normally the same effect.